client/cli/src/commands/purge_chain_cmd.rs

```diff
@@ -81,4 +92,8 @@ impl CliConfiguration for PurgeChainCmd {
 	fn shared_params(&self) -> &SharedParams {
 		&self.shared_params
 	}
+
+	fn database(&self) -> error::Result<Option<Database>> {
+		Ok(Some(self.database))
+	}
```

Comment:
```suggestion
	fn database(&self) -> Database {
		self.database
	}
```

You don't need the Result nor the Option here. The result is necessary for the CliConfiguration trait to allow the user to override and do some additional logical but for this private method you don't have to write it if you don't use it.

Reply:
But this is implementation of `CliConfiguration`, not a private method

Reply:
You don't need this function at all since you are not using it. You are using `config.database` and that one is set by the `import_params`. I think THIS is the major issue in this PR.

There are multiple solutions:
1. Either it makes sense to have the whole `import_params` in purge chain. If so, just add import_params and flatten the structopt. Then the database will be set properly (easiest)
2. If there are things in import_params that makes no sense whatsoever to have in purge chain, then you should probably split import params to have:
     - database_params (a new one you need to create with just what you need to configure the database)
     - import_params (which would include database_params)

I understand this is kinda complicated. I don't mind to take ownership on this PR if you don't want to get that much into sc-cli

Reply:
> But this is implementation of `CliConfiguration`, not a private method

Right! I didn't see with the code collapsing :sweat_smile: 

So I guess what is proposed here is working but has some duplicated code with the rest of the cli. You don't need the `cache_size` at all?

Reply:
If you want to refactor this further go ahead. I've also noticed that some other commands, such as `export-blocks` and `check-block` could use that parameter too. There needs to be some common place for DB config, which is not `ImportParams`. This is really of scope of this PR though.

Reply:
Ok I will take it from there then because I think it should be done for all the commands at once otherwise we would be tempted to do a similar temporary fixes on all the commands and it will stay that way forever.

Reply:
(CheckBlock is using ImportParams (which has the database) so I think that one is fine. I just had to fix ExportBlocks)